tests/Checker/Preparations/Use_Minimal_Theme_Preparation_Tests.php

```diff
@@ -0,0 +1,61 @@
+<?php
+/**
+ * Tests for the Check_Context class.
+ *
+ * @package plugin-check
+ */
+
+use WordPress\Plugin_Check\Checker\Preparation;
+use WordPress\Plugin_Check\Checker\Preparations\Use_Minimal_Theme_Preparation;
+
+class Use_Minimal_Theme_Preparation_Tests extends WP_UnitTestCase {
+	protected $theme_name;
+	protected $theme_slug;
+	protected $theme_dir;
+
+	public function set_up() {
+		parent::set_up();
+
+		$this->theme_name = 'WP Empty Theme';
+		$this->theme_slug = 'wp-empty-theme';
+		$this->theme_dir  = WP_PLUGIN_DIR . '/' . basename( TESTS_PLUGIN_DIR ) . '/test-content/themes';
+	}
+
+	public function test_get_theme_slug() {
+		$preparation = new Use_Minimal_Theme_Preparation( $this->theme_slug, $this->theme_dir );
+
+		$this->assertSame( $this->theme_slug, $preparation->get_theme_slug() );
+	}
+
+	public function test_prepare() {
+		$preparation = new Use_Minimal_Theme_Preparation( $this->theme_slug, $this->theme_dir );
+		$cleanup     = $preparation->prepare();
+
+		$this->assertIsCallable( $cleanup );
+		$this->assertSame( $this->theme_slug, get_option( 'template' ) );
+		$this->assertSame( $this->theme_slug, get_option( 'stylesheet' ) );
+		$this->assertSame( $this->theme_name, get_option( 'current_theme' ) );
+		$this->assertSame( str_replace( WP_CONTENT_DIR, '', $this->theme_dir ), get_option( 'template_root' ) );
+		$this->assertSame( str_replace( WP_CONTENT_DIR, '', $this->theme_dir ), get_option( 'stylesheet_root' ) );
+
+		$cleanup();
```

Comment:
The cleanup will not execute if any of the assertions fail. Can you assign all the call results to variables? This way you can then do the cleanup, and the assertions at the very end.

You already do something like that in the methods below, we need to do it here too.